File: db-4.8.30/csharp/LockStats.cs
/*-
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 2009 Oracle.  All rights reserved.
 *
 */
using System;
using System.Collections.Generic;
using System.Text;

namespace BerkeleyDB {
    /// <summary>
    /// Statistical information about the locking subsystem
    /// </summary>
    public class LockStats {
        private Internal.LockStatStruct st;
        internal LockStats(Internal.LockStatStruct stats) {
            st = stats;
        }

        /// <summary>
        /// Last allocated locker ID. 
        /// </summary>
        public uint LastAllocatedLockerID { get { return st.st_id; } }
        /// <summary>
        /// Lock conflicts w/ subsequent wait 
        /// </summary>
        public ulong LockConflictsWait { get { return st.st_lock_wait; } }
        /// <summary>
        /// Lock conflicts w/o subsequent wait 
        /// </summary>
        public ulong LockConflictsNoWait { get { return st.st_lock_nowait; } }
        /// <summary>
        /// Number of lock deadlocks. 
        /// </summary>
        public ulong LockDeadlocks { get { return st.st_ndeadlocks; } }
        /// <summary>
        /// Number of lock downgrades. 
        /// </summary>
        public ulong LockDowngrades { get { return st.st_ndowngrade; } }
        /// <summary>
        /// Number of lock modes. 
        /// </summary>
        public int LockModes { get { return st.st_nmodes; } }
        /// <summary>
        /// Number of lock puts. 
        /// </summary>
        public ulong LockPuts { get { return st.st_nreleases; } }
        /// <summary>
        /// Number of lock gets. 
        /// </summary>
        public ulong LockRequests { get { return st.st_nrequests; } }
        /// <summary>
        /// Number of lock steals so far. 
        /// </summary>
        public ulong LockSteals { get { return st.st_locksteals; } }
        /// <summary>
        /// Lock timeout. 
        /// </summary>
        public uint LockTimeoutLength { get { return st.st_locktimeout; } }
        /// <summary>
        /// Number of lock timeouts. 
        /// </summary>
        public ulong LockTimeouts { get { return st.st_nlocktimeouts; } }
        /// <summary>
        /// Number of lock upgrades. 
        /// </summary>
        public ulong LockUpgrades { get { return st.st_nupgrade; } }
        /// <summary>
        /// Locker lock granted without wait. 
        /// </summary>
        public ulong LockerNoWait { get { return st.st_lockers_nowait; } }
        /// <summary>
        /// Locker lock granted after wait. 
        /// </summary>
        public ulong LockerWait { get { return st.st_lockers_wait; } }
        /// <summary>
        /// Current number of lockers. 
        /// </summary>
        public uint Lockers { get { return st.st_nlockers; } }
        /// <summary>
        /// Current number of locks. 
        /// </summary>
        public uint Locks { get { return st.st_nlocks; } }
        /// <summary>
        /// Max length of bucket. 
        /// </summary>
        public uint MaxBucketLength { get { return st.st_hash_len; } }
        /// <summary>
        /// Maximum number steals in any partition. 
        /// </summary>
        public ulong MaxLockSteals { get { return st.st_maxlsteals; } }
        /// <summary>
        /// Maximum number of lockers so far. 
        /// </summary>
        public uint MaxLockers { get { return st.st_maxnlockers; } }
        /// <summary>
        /// Maximum num of lockers in table. 
        /// </summary>
        public uint MaxLockersInTable { get { return st.st_maxlockers; } }
        /// <summary>
        /// Maximum number of locks so far. 
        /// </summary>
        public uint MaxLocks { get { return st.st_maxnlocks; } }
        /// <summary>
        /// Maximum number of locks in any bucket. 
        /// </summary>
        public uint MaxLocksInBucket { get { return st.st_maxhlocks; } }
        /// <summary>
        /// Maximum number of locks in table. 
        /// </summary>
        public uint MaxLocksInTable { get { return st.st_maxlocks; } }
        /// <summary>
        /// Maximum number of steals in any partition. 
        /// </summary>
        public ulong MaxObjectSteals { get { return st.st_maxosteals; } }
        /// <summary>
        /// Maximum number of objects so far. 
        /// </summary>
        public uint MaxObjects { get { return st.st_maxnobjects; } }
        /// <summary>
        /// Maximum number of objectsin any bucket. 
        /// </summary>
        public uint MaxObjectsInBucket { get { return st.st_maxhobjects; } }
        /// <summary>
        /// Max partition lock granted without wait. 
        /// </summary>
        public ulong MaxPartitionLockNoWait { get { return st.st_part_max_nowait; } }
        /// <summary>
        /// Max partition lock granted after wait. 
        /// </summary>
        public ulong MaxPartitionLockWait { get { return st.st_part_max_wait; } }
        /// <summary>
        /// Current maximum unused ID. 
        /// </summary>
        public uint MaxUnusedID { get { return st.st_cur_maxid; } }
        /// <summary>
        /// Maximum num of objects in table. 
        /// </summary>
        public uint MaxObjectsInTable { get { return st.st_maxobjects; } }
        /// <summary>
        /// number of partitions. 
        /// </summary>
        public uint nPartitions { get { return st.st_partitions; } }
        /// <summary>
        /// Object lock granted without wait. 
        /// </summary>
        public ulong ObjectNoWait { get { return st.st_objs_nowait; } }
        /// <summary>
        /// Number of objects steals so far. 
        /// </summary>
        public ulong ObjectSteals { get { return st.st_objectsteals; } }
        /// <summary>
        /// Object lock granted after wait. 
        /// </summary>
        public ulong ObjectWait { get { return st.st_objs_wait; } }
        /// <summary>
        /// Current number of objects. 
        /// </summary>
        public uint Objects { get { return st.st_nobjects; } }
        /// <summary>
        /// Partition lock granted without wait. 
        /// </summary>
        public ulong PartitionLockNoWait { get { return st.st_part_nowait; } }
        /// <summary>
        /// Partition lock granted after wait. 
        /// </summary>
        public ulong PartitionLockWait { get { return st.st_part_wait; } }
        /// <summary>
        /// Region lock granted without wait. 
        /// </summary>
        public ulong RegionNoWait { get { return st.st_region_nowait; } }
        /// <summary>
        /// Region size. 
        /// </summary>
        public ulong RegionSize { get { return (ulong)st.st_regsize.ToInt64(); } }
        /// <summary>
        /// Region lock granted after wait. 
        /// </summary>
        public ulong RegionWait { get { return st.st_region_wait; } }
        /// <summary>
        /// Transaction timeout. 
        /// </summary>
        public uint TxnTimeoutLength { get { return st.st_txntimeout; } }
        /// <summary>
        /// Number of transaction timeouts. 
        /// </summary>
        public ulong TxnTimeouts { get { return st.st_ntxntimeouts; } }
        
    }
}

File: db-4.8.30/java/src/com/sleepycat/db/LockStats.java
/*-
 * Automatically built by dist/s_java_stat.
 * Only the javadoc comments can be edited.
 *
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 2002-2009 Oracle.  All rights reserved.
 */

package com.sleepycat.db;

/**
Lock statistics for a database environment.
*/
public class LockStats {
    // no public constructor
    /* package */ LockStats() {}

    private int st_id;
    /** TODO */
    public int getId() {
        return st_id;
    }

    private int st_cur_maxid;
    /** TODO */
    public int getCurMaxId() {
        return st_cur_maxid;
    }

    private int st_maxlocks;
    /** TODO */
    public int getMaxLocks() {
        return st_maxlocks;
    }

    private int st_maxlockers;
    /** TODO */
    public int getMaxLockers() {
        return st_maxlockers;
    }

    private int st_maxobjects;
    /** TODO */
    public int getMaxObjects() {
        return st_maxobjects;
    }

    private int st_partitions;
    /** TODO */
    public int getPartitions() {
        return st_partitions;
    }

    private int st_nmodes;
    /** TODO */
    public int getNumModes() {
        return st_nmodes;
    }

    private int st_nlockers;
    /** TODO */
    public int getNumLockers() {
        return st_nlockers;
    }

    private int st_nlocks;
    /** TODO */
    public int getNumLocks() {
        return st_nlocks;
    }

    private int st_maxnlocks;
    /** TODO */
    public int getMaxNlocks() {
        return st_maxnlocks;
    }

    private int st_maxhlocks;
    /** TODO */
    public int getMaxHlocks() {
        return st_maxhlocks;
    }

    private long st_locksteals;
    /** TODO */
    public long getLocksteals() {
        return st_locksteals;
    }

    private long st_maxlsteals;
    /** TODO */
    public long getMaxLsteals() {
        return st_maxlsteals;
    }

    private int st_maxnlockers;
    /** TODO */
    public int getMaxNlockers() {
        return st_maxnlockers;
    }

    private int st_nobjects;
    /** TODO */
    public int getNobjects() {
        return st_nobjects;
    }

    private int st_maxnobjects;
    /** TODO */
    public int getMaxNobjects() {
        return st_maxnobjects;
    }

    private int st_maxhobjects;
    /** TODO */
    public int getMaxHobjects() {
        return st_maxhobjects;
    }

    private long st_objectsteals;
    /** TODO */
    public long getObjectsteals() {
        return st_objectsteals;
    }

    private long st_maxosteals;
    /** TODO */
    public long getMaxOsteals() {
        return st_maxosteals;
    }

    private long st_nrequests;
    /** TODO */
    public long getNumRequests() {
        return st_nrequests;
    }

    private long st_nreleases;
    /** TODO */
    public long getNumReleases() {
        return st_nreleases;
    }

    private long st_nupgrade;
    /** TODO */
    public long getNumUpgrade() {
        return st_nupgrade;
    }

    private long st_ndowngrade;
    /** TODO */
    public long getNumDowngrade() {
        return st_ndowngrade;
    }

    private long st_lock_wait;
    /** TODO */
    public long getLockWait() {
        return st_lock_wait;
    }

    private long st_lock_nowait;
    /** TODO */
    public long getLockNowait() {
        return st_lock_nowait;
    }

    private long st_ndeadlocks;
    /** TODO */
    public long getNumDeadlocks() {
        return st_ndeadlocks;
    }

    private int st_locktimeout;
    /** TODO */
    public int getLockTimeout() {
        return st_locktimeout;
    }

    private long st_nlocktimeouts;
    /** TODO */
    public long getNumLockTimeouts() {
        return st_nlocktimeouts;
    }

    private int st_txntimeout;
    /** TODO */
    public int getTxnTimeout() {
        return st_txntimeout;
    }

    private long st_ntxntimeouts;
    /** TODO */
    public long getNumTxnTimeouts() {
        return st_ntxntimeouts;
    }

    private long st_part_wait;
    /** TODO */
    public long getPartWait() {
        return st_part_wait;
    }

    private long st_part_nowait;
    /** TODO */
    public long getPartNowait() {
        return st_part_nowait;
    }

    private long st_part_max_wait;
    /** TODO */
    public long getPartMaxWait() {
        return st_part_max_wait;
    }

    private long st_part_max_nowait;
    /** TODO */
    public long getPartMaxNowait() {
        return st_part_max_nowait;
    }

    private long st_objs_wait;
    /** TODO */
    public long getObjsWait() {
        return st_objs_wait;
    }

    private long st_objs_nowait;
    /** TODO */
    public long getObjsNowait() {
        return st_objs_nowait;
    }

    private long st_lockers_wait;
    /** TODO */
    public long getLockersWait() {
        return st_lockers_wait;
    }

    private long st_lockers_nowait;
    /** TODO */
    public long getLockersNowait() {
        return st_lockers_nowait;
    }

    private long st_region_wait;
    /** TODO */
    public long getRegionWait() {
        return st_region_wait;
    }

    private long st_region_nowait;
    /** TODO */
    public long getRegionNowait() {
        return st_region_nowait;
    }

    private int st_hash_len;
    /** TODO */
    public int getHashLen() {
        return st_hash_len;
    }

    private int st_regsize;
    /** TODO */
    public int getRegSize() {
        return st_regsize;
    }

    /**
    For convenience, the LockStats class has a toString method
    that lists all the data fields.
    */
    public String toString() {
        return "LockStats:"
            + "\n  st_id=" + st_id
            + "\n  st_cur_maxid=" + st_cur_maxid
            + "\n  st_maxlocks=" + st_maxlocks
            + "\n  st_maxlockers=" + st_maxlockers
            + "\n  st_maxobjects=" + st_maxobjects
            + "\n  st_partitions=" + st_partitions
            + "\n  st_nmodes=" + st_nmodes
            + "\n  st_nlockers=" + st_nlockers
            + "\n  st_nlocks=" + st_nlocks
            + "\n  st_maxnlocks=" + st_maxnlocks
            + "\n  st_maxhlocks=" + st_maxhlocks
            + "\n  st_locksteals=" + st_locksteals
            + "\n  st_maxlsteals=" + st_maxlsteals
            + "\n  st_maxnlockers=" + st_maxnlockers
            + "\n  st_nobjects=" + st_nobjects
            + "\n  st_maxnobjects=" + st_maxnobjects
            + "\n  st_maxhobjects=" + st_maxhobjects
            + "\n  st_objectsteals=" + st_objectsteals
            + "\n  st_maxosteals=" + st_maxosteals
            + "\n  st_nrequests=" + st_nrequests
            + "\n  st_nreleases=" + st_nreleases
            + "\n  st_nupgrade=" + st_nupgrade
            + "\n  st_ndowngrade=" + st_ndowngrade
            + "\n  st_lock_wait=" + st_lock_wait
            + "\n  st_lock_nowait=" + st_lock_nowait
            + "\n  st_ndeadlocks=" + st_ndeadlocks
            + "\n  st_locktimeout=" + st_locktimeout
            + "\n  st_nlocktimeouts=" + st_nlocktimeouts
            + "\n  st_txntimeout=" + st_txntimeout
            + "\n  st_ntxntimeouts=" + st_ntxntimeouts
            + "\n  st_part_wait=" + st_part_wait
            + "\n  st_part_nowait=" + st_part_nowait
            + "\n  st_part_max_wait=" + st_part_max_wait
            + "\n  st_part_max_nowait=" + st_part_max_nowait
            + "\n  st_objs_wait=" + st_objs_wait
            + "\n  st_objs_nowait=" + st_objs_nowait
            + "\n  st_lockers_wait=" + st_lockers_wait
            + "\n  st_lockers_nowait=" + st_lockers_nowait
            + "\n  st_region_wait=" + st_region_wait
            + "\n  st_region_nowait=" + st_region_nowait
            + "\n  st_hash_len=" + st_hash_len
            + "\n  st_regsize=" + st_regsize
            ;
    }
}
